src/Microsoft.ApplicationInsights/TelemetryClient.cs

```diff
@@ -516,44 +518,95 @@ public void Initialize(ITelemetry telemetry)
             }
 
             telemetry.Context.Initialize(this.Context, instrumentationKey);
-            foreach (ITelemetryInitializer initializer in this.configuration.TelemetryInitializers)
+
+            ISupportAdvancedSampling telemetryWithSampling = telemetry as ISupportAdvancedSampling;
+
+            // Telemetry can be already sampled out if that decision was made before calling Track()
+            bool sampledOut = telemetryWithSampling?.IsProactivelySampledOut ?? false;
+            bool shouldTryHeadSampling = this.proactiveSamplingEnabled && (telemetryWithSampling?.SupportsProactiveSampling ?? false);
+
+            if (!sampledOut)
             {
-                try
+                if (shouldTryHeadSampling)
                 {
-                    initializer.Initialize(telemetry);
+                    // If telemetry was not sampled out, we need to check its sampling score after we learn its Operation Id                    
```

Comment:
can we be more aggresive and move lines 492 to 518 to inside - if(shouldTryHeadSampling) ? We don't want to pay the price of doing ConcurrentDictionary instantiation and copying, if item is to be proactively sampled out.

Reply:
It didn't show up in perf trace because the scenario was not triggered.
The following copies all context from TelemetryClitn to individual telemetry, instantiating a concurrent dictionary if needed. This can be avoided if item is being sampled out.
Utils.CopyDictionary(this.Context.Properties, telemetry.Context.Properties);

(there is nothing preventing an autocollector (eg: aspnet core request collection) to use telemetryclient.context)

Let me know if this make sense/not.